Slim/Networking/Async/HTTP.pm

```diff
@@ -107,6 +109,9 @@ sub new_socket {
 		);
 	}
 
+	# merge with user-defined socket parameters
+	push @_, %{$self->options} if $self->options;
+
```

Comment:
Doesn't this belong to before the proxy check? Otherwise a proxied call would miss these additional parameters, wouldn't it?

Reply:
I agree, but I was not sure that it would be needed anyway when there is a proxy, I can move it, it probably does not cost much anyway

Reply:
Although, after re-reading IO::Socket::INET
```
PeerAddr    Remote host address          <hostname>[:<port>]
PeerHost    Synonym for PeerAddr
PeerPort    Remote port or service       <service>[(<no>)] | <no>
LocalAddr   Local host bind address      hostname[:port]
LocalHost   Synonym for LocalAddr
LocalPort   Local host bind port         <service>[(<no>)] | <no>
Proto       Protocol name (or number)    "tcp" | "udp" | ...
Type        Socket type              SOCK_STREAM | SOCK_DGRAM | ...
Listen      Queue size for listen
ReuseAddr   Set SO_REUSEADDR before binding
Reuse       Set SO_REUSEADDR before binding (deprecated,
                                             prefer ReuseAddr)
ReusePort   Set SO_REUSEPORT before binding
Broadcast   Set SO_BROADCAST before binding
Timeout     Timeout value for various operations
MultiHomed  Try all addresses for multi-homed hosts
Blocking    Determine if connection will be blocking mode
```
I'm not sure there is anything there that can be useful for user to change. SSL is a different story but when proxy is involved, this is out of control of the caller